src/libraries/System.Net.Http/src/System/Net/Http/StreamToStreamCopy.cs

```diff
@@ -16,6 +16,24 @@ namespace System.Net.Http
     /// </summary>
     internal static class StreamToStreamCopy
     {
+        /// <summary>Copies the source stream from its current position to the destination stream at its current position.</summary>
+        /// <param name="source">The source stream from which to copy.</param>
+        /// <param name="destination">The destination stream to which to copy.</param>
+        /// <param name="bufferSize">The size of the buffer to allocate if one needs to be allocated. If zero, use the default buffer size.</param>
+        /// <param name="disposeSource">Whether to dispose of the source stream after the copy has finished successfully.</param>
+        public static void Copy(Stream source, Stream destination, int bufferSize, bool disposeSource)
+        {
+            Debug.Assert(source != null);
+            Debug.Assert(destination != null);
+            Debug.Assert(bufferSize >= 0);
+
+            source.CopyTo(destination, bufferSize);
```

Comment:
Partial mitigation for long copies: This should do incremental reads & writes and check for cancelation in between.